library/entity/multipart/src/main/java/org/quiltmc/qsl/entity/multipart/api/EntityPart.java

```diff
@@ -30,4 +38,29 @@
 @InjectedInterface(EnderDragonPart.class)
 public interface EntityPart<E extends Entity> {
 	E getOwner();
+
+	/**
+	 * Renders the hitbox for the entity part.
+	 * <p>
+	 * Should normally not be overridden unless it is to more accurately draw non-standard hitboxes.
+	 * @param matrices the {@link MatrixStack matrix stack} used for rendering
+	 * @param vertices the {@link VertexConsumer vertex consumer} used for rendering
+	 * @param ownerX the {@link #getOwner() owner's} rendered X coordinate
+	 * @param ownerY the {@link #getOwner() owner's} rendered Y coordinate
+	 * @param ownerZ the {@link #getOwner() owner's} rendered Z coordinate
+	 * @param owner the {@link #getOwner() owner}
+	 * @param tickDelta progress for linearly interpolating between the previous and current game state
+	 */
+	@Environment(EnvType.CLIENT)
+	default void renderHitbox(MatrixStack matrices, VertexConsumer vertices, double ownerX, double ownerY, double ownerZ, Entity owner, float tickDelta) {
```

Comment:
Oh damn, a client-only method in a normal class?

We would scare some people with that :P
I'm not sure what other option we have though. Definitely means we won't be able to split QSL into common/client source sets, not that we were planning to.
Will take a bit to think about it, but don't take my comment as blocking.

Reply:
I think this is the best solution. We could create some kind of optional EntityPartRenderingRegistry, but I think this method is best for discoverability and ease of use.